eng/targets/Helix.targets

```diff
@@ -171,6 +176,21 @@
     <ConvertToAbsolutePath Paths="$(PublishDir)">
       <Output TaskParameter="AbsolutePaths" PropertyName="PublishAbsoluteDir" />
     </ConvertToAbsolutePath>
+
+    <!-- Another heavy hammer: Overwrite assemblies from work item payload with those found in shared Fx. -->
+    <ItemGroup>
+      <_FilesFromPublish Include="$(PublishAbsoluteDir)*.dll" />
+      <_FilesFromSharedFx Include="$(LocalDotNetRoot)shared\$(SharedFxName)\$(SharedFxVersion)\*.dll" />
+      <_FilesToOverwrite Include="@(_FilesFromSharedFx)"
+          Condition=" '%(Filename)' != '' AND '@(_FilesFromPublish)' != '' " />
+    </ItemGroup>
+    <Copy SourceFiles="@(_FilesToOverwrite)"
+        DestinationFolder="$(PublishAbsoluteDir)"
+        OverwriteReadOnlyFiles="$(OverwriteReadOnlyFiles)"
+        Retries="$(CopyRetryCount)"
+        RetryDelayMilliseconds="$(CopyRetryDelayMilliseconds)"
+        SkipUnchangedFiles="$(SkipCopyUnchangedFiles)" />
```

Comment:
> Does that imply that the failing test actually tried to run the binary on the .NET 4.6.1 runtime, and the succeeding test ran on the .NET 6 runtime.

Yes, that's almost definitely what it means. I didn't consider the TFM in this block. Should only `<Copy/>` when targeting `net6.0`. I'll add a commit or two to (a) focus temporarily on Windows ➡️ Windows and Linux ➡️ Linux jobs (because the other combinations are failing as consistently as we expect) and (b) fix this targeting issue.